src/runtime-rs/crates/hypervisor/src/kernel_param.rs

```diff
@@ -67,43 +69,37 @@ impl KernelParams {
         Self { params }
     }
 
-    pub(crate) fn new_rootfs_kernel_params(rootfs_driver: &str) -> Self {
-        let params = match rootfs_driver {
-            VM_ROOTFS_DRIVER_BLK => {
-                vec![
-                    Param {
-                        key: "root".to_string(),
-                        value: "/dev/vda1".to_string(),
-                    },
-                    Param {
-                        key: "rootflags".to_string(),
-                        value: "data=ordered,errors=remount-ro ro".to_string(),
-                    },
-                    Param {
-                        key: "rootfstype".to_string(),
-                        value: "ext4".to_string(),
-                    },
-                ]
+    pub(crate) fn new_rootfs_kernel_params(rootfs_driver: &str, rootfs_type: &str) -> Result<Self> {
+        let mut params = vec![];
+
+        if rootfs_driver == VM_ROOTFS_DRIVER_PMEM {
+            params.push(Param::new("root", VM_ROOTFS_ROOT_PMEM));
+            if rootfs_type == VM_ROOTFS_FILESYSTEM_EXT4 || rootfs_type == VM_ROOTFS_FILESYSTEM_XFS {
+                params.push(Param::new(
+                    "rootflags",
+                    "dax,data=ordered,errors=remount-ro ro",
+                ));
+            } else if rootfs_type == VM_ROOTFS_FILESYSTEM_EROFS {
+                params.push(Param::new("rootflags", "dax ro"));
+            } else {
+                return Err(anyhow!("Unsupported rootfs type"));
```

Comment:
Reduce the use of `if {} else if {}` and a more elegant way is `match` {...}

Reply:
@Apokleos Thanks for your review! I switched `if {} else if {}` to `match{...}`.